src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java

```diff
@@ -0,0 +1,138 @@
+
+	@Autowired
+	public UISampleService(SampleService sampleService, ProjectService projectService, UpdateSamplePermission updateSamplePermission,
+			SequencingObjectService sequencingObjectService, GenomeAssemblyService genomeAssemblyService, UICartService cartService) {
+		this.sampleService = sampleService;
+		this.projectService = projectService;
+		this.updateSamplePermission = updateSamplePermission;
+		this.sequencingObjectService = sequencingObjectService;
+		this.genomeAssemblyService = genomeAssemblyService;
+		this.cartService = cartService;
+	}
+
+	/**
+	 * Get full details, including metadata for a {@link Sample}
+	 *
+	 * @param id Identifier for a {@link Sample}
+	 * @return {@link SampleDetails}
+	 */
+	public SampleDetails getSampleDetails(Long id) {
+		Sample sample = sampleService.read(id);
+		Authentication authentication = SecurityContextHolder.getContext()
+				.getAuthentication();
+		boolean isModifiable = updateSamplePermission.isAllowed(authentication, sample);
+		return new SampleDetails(sample, isModifiable, cartService.isSampleInCart(id));
+	}
+
+	/**
+	 * Get the sequence files associated with a sample
+	 *
+	 * @param sampleId  Identifier for a sample
+	 * @param projectId Identifier for the project the sample belong to
+	 * @return All the sequencing files associated with the sample
+	 */
+	public SampleFiles getSampleFiles(Long sampleId, Long projectId) {
+		Sample sample = sampleService.read(sampleId);
+		// get the project if available
+		Project project = null;
+		if (projectId != null) {
+			project = projectService.read(projectId);
+		}
+
+		Collection<SampleSequencingObjectJoin> filePairJoins = sequencingObjectService.getSequencesForSampleOfType(
+				sample, SequenceFilePair.class);
+		Collection<SampleSequencingObjectJoin> singleFileJoins = sequencingObjectService.getSequencesForSampleOfType(
+				sample, SingleEndSequenceFile.class);
+		Collection<SampleSequencingObjectJoin> fast5FileJoins = sequencingObjectService.getSequencesForSampleOfType(
+				sample, Fast5Object.class);
+		List<SequencingObject> fast5 = fast5FileJoins.stream()
+				.map(SampleSequencingObjectJoin::getObject)
+				.collect(Collectors.toList());
+
+		Collection<SampleGenomeAssemblyJoin> genomeAssemblyJoins = genomeAssemblyService.getAssembliesForSample(sample);
+
+		List<GenomeAssembly> genomeAssemblies = genomeAssemblyJoins.stream()
+				.map(SampleGenomeAssemblyJoin::getObject)
+				.collect(Collectors.toList());
+
+		// add project to qc entries and filter any unavailable entries
+		List<SequencingObject> filePairs = new ArrayList<>();
+		for (SampleSequencingObjectJoin join : filePairJoins) {
+			SequencingObject obj = join.getObject();
+			enhanceQcEntries(obj, project);
+			filePairs.add(obj);
+		}
+
+		List<SequencingObject> singles = new ArrayList<>();
+		for (SampleSequencingObjectJoin join : singleFileJoins) {
+			SequencingObject obj = join.getObject();
+			enhanceQcEntries(obj, project);
+			singles.add(obj);
+		}
+
```

Comment:
Fast5 data also has qc files associated with it. So I think there should be a section here which sets the enhanceQCEntries for it

Reply:
It's not in the current version of the files.  I can to this in a later branch. Not actually doing anything with the QC data yet.

Reply:
Sounds good!